### PR TITLE
[@types/mongoose] Allow the use of "DocumentArray.discriminator(name, schema, value)"

### PR DESCRIPTION
**Pull request by hasezoey**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://mongoosejs.com/docs/api.html#documentarraypath_DocumentArrayPath-discriminator
  - https://github.com/Automattic/mongoose/blob/master/lib/schema/documentarray.js#L153
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongoose/index.d.ts [hunk 1]
```diff
@@ -2407,15 +2407,17 @@ declare module "mongoose" {
...
+        discriminator<U extends Document>(name: string, schema: Schema, value?: string): Model<U>;
 
         /**
          * Adds a discriminator type.
          * @param name discriminator model name
          * @param schema discriminator model schema
+         * @param value the string stored in the `discriminatorKey` property
          */
-        discriminator<U extends Document, M extends Model<U>>(name: string, schema: Schema): M;
+        discriminator<U extends Document, M extends Model<U>>(name: string, schema: Schema, value?: string): M;
```
Comment (ToucheSir) [inline types/mongoose/index.d.ts]: Could you add a test or two that covers this function signature change?

**Review by ToucheSir — CHANGES_REQUESTED** [review]

File: types/mongoose/index.d.ts (hunk 1 above)
Comment (hasezoey) [reply types/mongoose/index.d.ts]: Sorry, im new to this repo - do you mean mongoose code, or testing the typings?

Comment (hasezoey) [inline types/mongoose/index.d.ts]: should i rename "value" to "tiedValue" because that would be the name in the code, but "[value]" is used in the jsdoc

Comment (hasezoey) [reply types/mongoose/index.d.ts]: does 8d6981d8ada94d745a8941438a8d01661e776fab do what you asked?

Comment (ToucheSir) [reply types/mongoose/index.d.ts]: Yes, that's exactly it!

Comment (ToucheSir) [reply types/mongoose/index.d.ts]: It's probably fine as-is? Presumably people are going to consult the documentation before reading the actual implementation code.

**Review by ToucheSir — APPROVED** [review]

**Pull request merged**